src/battle-animations.ts

```diff
@@ -626,10 +626,10 @@ class BattleScene {
 				if (speciesOverage && speciesTable.includes(species)) {
 					for (const sidebarIcon of sidebarIcons) {
 						if (side.pokemon[sidebarIcon[1]!].getBaseTemplate().baseSpecies === species) {
-							sidebarIcon[0] = 'pokemon-illusion';
+							if (this.gen < 8) sidebarIcon[0] = 'pokemon-illusion';
```

Comment:
For one thing, why would you put an `if` this deep into an inner loop? You're running the loop for no reason if `this.gen >= 8`.